src/libs/SearchUtils.ts

```diff
@@ -655,26 +659,60 @@ function buildQueryStringFromFilterFormValues(filterValues: Partial<SearchAdvanc
 /**
  * returns the values of the filters in a format that can be used in the SearchAdvancedFiltersForm as initial form values
  */
-function buildFilterFormValuesFromQuery(queryJSON: SearchQueryJSON) {
-    const filters = getFilters(queryJSON);
+function buildFilterFormValuesFromQuery(
+    queryJSON: SearchQueryJSON,
+    policyCategories: OnyxCollection<OnyxTypes.PolicyCategories>,
+    policyTags: OnyxCollection<OnyxTypes.PolicyTagLists>,
+    currencyList: OnyxTypes.CurrencyList,
+    personalDetails: OnyxTypes.PersonalDetailsList,
+    cardList: OnyxTypes.CardList,
+    reports: OnyxCollection<OnyxTypes.Report>,
+    taxRates: Record<string, string[]>,
+) {
+    const filters = queryJSON.flatFilters;
     const filterKeys = Object.keys(filters);
     const filtersForm = {} as Partial<SearchAdvancedFiltersForm>;
+    const policyID = queryJSON.policyID;
     for (const filterKey of filterKeys) {
         if (filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.REPORT_ID || filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.MERCHANT || filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.DESCRIPTION) {
             filtersForm[filterKey] = filters[filterKey]?.[0]?.value.toString();
         }
-        if (
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.CATEGORY ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.TAX_RATE ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPENSE_TYPE ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.TAG ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.CURRENCY ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.FROM ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.TO ||
-            filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.IN
-        ) {
-            filtersForm[filterKey] = filters[filterKey]?.map((filter) => filter.value.toString());
+        if (filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPENSE_TYPE) {
+            filtersForm[filterKey] = filters[filterKey]
+                ?.map((expenseType) => expenseType.value.toString())
+                .filter((expenseType) => Object.values(CONST.SEARCH.TRANSACTION_TYPE).includes(expenseType as ValueOf<typeof CONST.SEARCH.TRANSACTION_TYPE>));
+        }
+        if (filterKey === CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID) {
+            filtersForm[filterKey] = filters[filterKey]?.map((card) => card.value.toString()).filter((card) => Object.keys(cardList).includes(card));
```

Comment:
Same here and for the filters below
```suggestion
            filtersForm[filterKey] = filters[filterKey]?.map((card) => card.value.toString()).filter((card) => !!cardList[card]);
```